neuro_evolution_ctrnn/brains/i_layer_based_brain.py

```diff
@@ -0,0 +1,190 @@
+import abc
+from brains.i_brain import IBrain, ConfigClass
+from gym import Space
+import numpy as np
+from tools.configurations import ILayerBasedBrainCfg
+from typing import List, TypeVar, Generic
+
+LayerdConfigClass = TypeVar('LayerdConfigClass', bound=ILayerBasedBrainCfg)
+
+
+class ILayerBasedBrain(IBrain, Generic[LayerdConfigClass]):
+
+    @abc.abstractmethod
+    def __init__(self, input_space: Space, output_space: Space, individual: np.ndarray, config: LayerdConfigClass):
+        super().__init__(input_space, output_space, individual, config)
+        input_size = self._size_from_space(input_space)
+        output_size = self._size_from_space(output_space)
+        hidden_structure: List[int] = config.hidden_structure
+
+        # initialize weights out of individual
+
+        individual_index = 0  # progress index
+        # initialize emphty
+        self.weight_ih = []  # Weights for weighted input values
+        self.weight_hh = []  # Weights for weighted stored values
+        self.bias_h = []  # Biases
+        self.hidden = []  # Initial values for state storage
+        self.layer_output = []  # Weights from output last Layer to output nodes
+        number_gates = self.get_number_gates()
+
+        # iterate for all given layers in the structure
+        for layer in range(len(hidden_structure)):
+
+            # Matrices for weighted input values in calculations
+
+            if layer == 0:  # The first Layer don't has an output from the previous layer, but the input values
+                self.weight_ih.append(
+                    np.array(
+                        individual[
+                            individual_index:
+                            individual_index + number_gates * hidden_structure[layer] * input_size
+                        ]
+                    ).reshape((number_gates, hidden_structure[layer], input_size))
+                )
+                individual_index += number_gates * input_size * hidden_structure[0]
```

Comment:
in Zeile 41 steht `hidden_structure[layer]` und hier steht `hidden_structure[0]` für die gleiche Sache. Bitte einheitlich machen. 

Und wenn man schon dabei ist, kann man auch gleich DRY machen und den code vereinfachen. 

Beachte: Wenn hier etwas an der Form geändert wird, muss es bei den nachfollgenden zeilen auch angepasst werden, damit es einheitlich bleibt. 

```suggestion
                number_elements = individual_index + number_gates * hidden_structure[layer] * input_size
                self.weight_ih.append(np.array(
                        individual[individual_index: number_elements+number_elements]
                    ).reshape((number_gates, hidden_structure[layer], input_size))
                )
                individual_index += number_elements
```

Reply:
erledigt